### PR TITLE
Wenxuan

### PR DESCRIPTION
**Pull request by Jeslin1231**
Balance Display and Profile Page

### CODE REVIEW

File: src/components/PayinComfirmation.jsx [hunk 1]
```diff
@@ -81,6 +81,23 @@ const PayinComfirmation = () => {
...
+      // alert(`Your balance is: ${balance} USDT`);
+      setWalletBalance(balance);
+    }
+  })
+  .catch((error) => {
+    console.error("Error:", error);
+  });
+  }, [checkBalance]);
+
+
```
Comment (corazhangjob) [inline src/components/PayinComfirmation.jsx:100]: `          <a herf="" className="underline underline-offset-1 italic">
            MoonPay
          </a>`
Line 106 -- The herf attribute in the anchor tag should be corrected to href.

File: src/components/UpdateCredit.jsx [hunk 2]
```diff
@@ -1,21 +1,36 @@
...
 const UpdateCredit = () => {
   const { currentColor } = useStateContext();
-  const [cardNumber, setCardNumber] = useState("111 ");
+  const [cardNumber, setCardNumber] = useState("000");
+  const dispatch = useDispatch(); 
   const stripe = useStripe();
   const elements = useElements();
   const JWTtoken = localStorage.getItem("jwtToken");
+
+  useEffect(() => {
```
Comment (corazhangjob) [inline src/components/UpdateCredit.jsx:23]: There is another unused useEffect hook towards the end of the component, consider removing it if it's not needed.

Comment (Jeslin1231) [reply src/components/UpdateCredit.jsx:23]: Removed, thanks for pointing out

File: src/components/PayinComfirmation.jsx (hunk 1 above)
Comment (Jeslin1231) [reply src/components/PayinComfirmation.jsx:100]: Corrected

File: src/components/Deposit.js [hunk 3]
```diff
@@ -1,540 +1,536 @@
...
+  },
 
+  getUserWalletBalance: async () => {
+    const usdtContract = new ethers.Contract(usdtAddress, usdtAbi, signer);
+    try {
+      const userAddress = await signer.getAddress();
+      const balance = await usdtContract.balanceOf(userAddress);
+
+      // USDT has 6 decimals, so you may need to divide the balance by 1e6
+      const formattedBalance = ethers.utils.formatUnits(balance, 18);
```
Comment (corazhangjob) [inline src/components/Deposit.js:527]: The comment about USDT having 6 decimals is noted, but the code uses 18 decimals when calling ethers.utils.formatUnits.

Comment (Jeslin1231) [reply src/components/Deposit.js:527]: This comment is from code provided by Danny, USDT actually has 18 decimal after I tried for multiple times, I will correct the wrong comment.

Comment (Jeslin1231) [reply src/components/Deposit.js:527]: Corrected.

**Pull request merged**